docs/dev/table/streaming/unbounded-data-processing/joins.md

```diff
@@ -351,4 +339,20 @@ FROM
 
 <span class="label label-danger">Attention</span> Flink does not support event time temporal table joins currently.
 
+### Temporal Table joins vs Other joins
+
+In contrast to [regular joins](#regular-joins), the previous results of the temporal table join will not be affected despite the changes on the build side. Also, the temporal table join operator is very lightweight and does not keep any state.
+
+Compared to [interval joins](#interval-joins), temporal table joins do not define a time window within which the records will be joined.
```

Comment:
again, 1.12 added support for temporal event time table joins so this needs to be updated.